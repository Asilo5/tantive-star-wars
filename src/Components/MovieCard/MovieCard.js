import React from 'react';
import { Link } from 'react-router-dom';
import './MovieCard.scss';

const MovieCard = ({title, episode_id, release_date, id, characters, openingCrawl, setCharacters, addMovieScroll }) => {
    return ( 
    <article className="movieCard">
      <ul>    
        <li>{title}</li>
        <li>Episode: {episode_id}</li>
        <li>Released in {release_date}</li>
        <Link to='/movies/:id'>
        <button 
          className="movie_card_btn" 
          type="submit" 
          // id={episode_id}
          onClick={() => {
            setCharacters(characters)
            addMovieScroll(openingCrawl)
            }}>View Characters
        </button>
        </Link>
      </ul>
    </article>
    )
}

export default MovieCard;