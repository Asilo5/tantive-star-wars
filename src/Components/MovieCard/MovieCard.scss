.movieCard {
    background-color:#00ffff9e;
    border-radius: 3px;
    box-shadow: 0px 1px whitesmoke, 0 0 1px whitesmoke, 0 0 1em limegreen;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    height: 193px;
    margin: 1%;
    width: 28%;
}

ul {
    font-size: 1.2em;
    list-style: none;
    padding: 0;
}

.movie_card_btn {
    font-size: 1em;
    height: 30px;
    margin: 9%;
    &:hover {
        font-weight: 800;
        cursor: pointer;
        transform: scale(1.1);
        transition: all .1s ease-in-out;
      }
}