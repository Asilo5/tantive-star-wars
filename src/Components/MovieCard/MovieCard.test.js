import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {

    let wrapper = shallow(<MovieCard />)

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});