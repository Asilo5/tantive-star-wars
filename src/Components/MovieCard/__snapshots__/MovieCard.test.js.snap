// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot with all information passing in correctly 1`] = `
<article
  className="movieCard"
>
  <ul>
    <li />
    <li>
      Episode: 
    </li>
    <li>
      Released in 
    </li>
    <Link
      to="/characters"
    >
      <button
        className="movie_card_btn"
        onClick={[Function]}
        type="submit"
      >
        View Characters
      </button>
    </Link>
  </ul>
</article>
`;
